mesonbuild/modules/python.py

```diff
@@ -0,0 +1,320 @@
+            lib = Path(self.variables.get('LIBDIR')) / libname
+        if not lib.exists():
+            mlog.log('Could not find Python3 library {!r}'.format(str(lib)))
+            return None
+        return [str(lib)]
+
+    def _find_libpy_windows(self, env):
+        '''
+        Find python3 libraries on Windows and also verify that the arch matches
+        what we are building for.
+        '''
+        pyarch = self.get_windows_python_arch()
+        if pyarch is None:
+            self.is_found = False
+            return
+        arch = detect_cpu_family(env.coredata.compilers)
+        if arch == 'x86':
+            arch = '32'
+        elif arch == 'x86_64':
+            arch = '64'
+        else:
+            # We can't cross-compile Python 3 dependencies on Windows yet
+            mlog.log('Unknown architecture {!r} for'.format(arch),
+                     mlog.bold(self.name))
+            self.is_found = False
+            return
+        # Pyarch ends in '32' or '64'
+        if arch != pyarch:
+            mlog.log('Need', mlog.bold(self.name), 'for {}-bit, but '
+                     'found {}-bit'.format(arch, pyarch))
+            self.is_found = False
+            return
+        # This can fail if the library is not found
+        largs = self.get_windows_link_args()
+        if largs is None:
+            self.is_found = False
+            return
+        self.link_args = largs
+        # Compile args
+        inc = self.paths.get('include')
+        self.compile_args = ['-I' + inc]
+        platinc = self.paths.get('platinclude')
+        if platinc and inc != platinc:
+            self.compile_args.append('-I' + platinc)
+        self.is_found = True
+
+    @staticmethod
+    def get_methods():
+        if mesonlib.is_windows():
+            return [DependencyMethods.PKGCONFIG, DependencyMethods.SYSCONFIG]
+        elif mesonlib.is_osx():
+            return [DependencyMethods.PKGCONFIG, DependencyMethods.EXTRAFRAMEWORK]
+        else:
+            return [DependencyMethods.PKGCONFIG]
+
+    def get_pkgconfig_variable(self, variable_name, kwargs):
+        if self.pkgdep:
+            return self.pkgdep.get_pkgconfig_variable(variable_name, kwargs)
+        else:
+            return super().get_pkgconfig_variable(variable_name, kwargs)
+
+
+class PythonHolder(ExternalProgramHolder, InterpreterObject):
+    def __init__(self, interpreter, python):
+        InterpreterObject.__init__(self)
+        ExternalProgramHolder.__init__(self, python)
+        self.interpreter = interpreter
+        prefix = self.interpreter.environment.coredata.get_builtin_option('prefix')
+        self.variables = eval(run_command(python, "import sysconfig; print (repr(sysconfig.get_config_vars()))"))
+        self.paths = eval(run_command(python, "import sysconfig; print (repr(sysconfig.get_paths(" \
+                "vars={'base': '%s', 'platbase': '%s', 'installed_base': '%s'})))" % (prefix, prefix, prefix)))
```

Comment:
__[Flake8]__

[E128] continuation line under-indented for visual indent


<!-- Comment by [SideCI](https://sideci.com) -->